actions.py

```diff
@@ -465,35 +649,66 @@ def submit(self, dispatcher: CollectingDispatcher,
         departure_iata = tracker.get_slot("iata_departure")
         destination_airport = tracker.get_slot("travel_destination")
         destination_iata = tracker.get_slot("iata_destination")
-
-        message = f"A one-way flight from {departure_airport} ({departure_iata}) to {destination_airport} ({destination_iata}) "\
-                  f"emits {co2_in_tons} of CO2. " \
-                  f"It would be amazing if you bought offsets for that carbon! " \
-                  f"There are some great, UN-certified projects you can pick from."
+        stopover_airport = tracker.get_slot("travel_stopover")
+        stopover_iata = tracker.get_slot("iata_stopover")
+        previous_entered_flight = tracker.get_slot("previous_entered_flight")
+
+        if not stopover_iata:
+            message = (
+                f"A one-way flight from {departure_airport} ({departure_iata}) to {destination_airport} ({destination_iata}) "
+                f"emits {co2_in_tons[0]} of CO2. "
+                f"It would be amazing if you bought offsets for that carbon! "
+                f"There are some great, UN-certified projects you can pick from."
+            )
+        else:
+            message = (
+                f"The trip from {departure_airport} ({departure_iata}) to {destination_airport} ({destination_iata}) "
+                f"via {stopover_airport} ({stopover_iata}) emits {float(co2_in_tons[0].split()[0])+float(co2_in_tons[1].split()[0]):.1f} {co2_in_tons[0].split()[1]} of CO2. "
+                f"The first leg emits {co2_in_tons[0]} of CO2. "
+                f"The second leg emits {co2_in_tons[1]} of CO2."
+                f"It would be amazing if you bought offsets for that carbon! "
+                f"There are some great, UN-certified projects you can pick from."
+            )
 
         if tracker.get_latest_input_channel() == "facebook":
             payload = hyperlink_payload(tracker, message, "Buy Offsets", url)
             dispatcher.utter_custom_json(payload)
         else:
             dispatcher.utter_message(message + f" [Buy Offsets]({url})")
-        return [
-            SlotSet("travel_departure"),
-            SlotSet("iata_departure"),
-            SlotSet("travel_destination"),
-            SlotSet("iata_destination"),
-            SlotSet("travel_flight_class")
-        ]
 
+        if tracker.get_slot("travel_stopover"):
+            return [
+                SlotSet("travel_departure"),
+                SlotSet("iata_departure"),
+                SlotSet("travel_destination"),
+                SlotSet("iata_destination"),
+                SlotSet("travel_stopover"),
+                SlotSet("iata_stopover"),
+                SlotSet("travel_flight_class"),
+                SlotSet("previous_entered_flight"),
```

Comment:
This is the same list as the next, except with one extra element. To make it more clear, first create the list that both have in common, and then write something like `if ...: return unset_travel_slots else: return unset_travel_slots + [SlotSet("previous_entered_flight")]`